src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -105,17 +105,18 @@ public Task InitializeAsync(ProtectedPrerenderComponentApplicationStore store, C
                     await OnCircuitOpenedAsync(cancellationToken);
                     await OnConnectionUpAsync(cancellationToken);
 
-                    // We add the root components *after* the circuit is flagged as open.
-                    // That's because AddComponentAsync waits for quiescence, which can take
-                    // arbitrarily long. In the meantime we might need to be receiving and
-                    // processing incoming JSInterop calls or similar.
```

Comment:
Now that this isn't the case, would there be any reason to change it so root components are added *before* the circuit is flagged as open?

Reply:
I think it still is the case. JS interop calls can happen at any time, including while we're waiting for the first (or all) root component(s) to be added. Or am I misunderstanding your point?

Reply:
I was more just unsure if this comment was implying that the only reason we're adding root components after the circuit is flagged as open is that `AddComponentAsync` waits for quiescence. So I guess I was wondering, if we're not immediately awaiting the `Task` returned by `AddComponentAsync`, and instead awaiting them all later (via `Task.WhenAll`), would there be any reason then to put the "flag the circuit as open" step in between adding the components and waiting for quiescence? e.g.:

```csharp
for (var i = 0; i < count; i++)
{
  // Call `AddComponentAsync` and collect its task.
}

// Flag the circuit as open.
await OnCircuitOpenedAsync(cancellationToken);
await OnConnectionUpAsync(cancellationToken);

// Await quiescence.
await Task.WhenAll(...);
```

I personally don't see a reason why we would need or want to do this, but I'm just bringing it up because I saw that code comment and was unsure if it was implying something like this would be desirable somehow.

Reply:
I'm entirely expecting the answer to be "no, we shouldn't do this", but I just want to make sure there's not something I'm glancing over.

Reply:
There could be a back-compat argument for keeping the previous ordering, since developers would have been able to observe their OnCircuitOpenedAsync/OnConnectionUpAsync callbacks firing before their component constructors.

If there was a strong reason for changing it we probably could do, but since it seems equally valid either way, I would lean towards retaining the ordering. If you come up with any reason why you think it really should be changed then let us know!